src/docker/build.rs

```diff
@@ -0,0 +1,121 @@
+use std::env;
+use std::process::Command;
+use std::str::FromStr;
+
+use super::engine::Engine;
+use crate::errors::*;
+use crate::shell::Verbosity;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Progress {
+    Plain,
+    Auto,
+    Tty,
+}
+
+impl FromStr for Progress {
+    type Err = eyre::ErrReport;
+
+    fn from_str(progress: &str) -> Result<Self> {
+        Ok(match progress {
+            "plain" => Progress::Plain,
+            "auto" => Progress::Auto,
+            "tty" => Progress::Tty,
+            s => eyre::bail!("unexpect progress type: expected plain, auto, or tty and got {s}"),
+        })
+    }
+}
+
+impl From<Progress> for &str {
+    fn from(progress: Progress) -> Self {
+        match progress {
+            Progress::Plain => "plain",
+            Progress::Auto => "auto",
+            Progress::Tty => "tty",
+        }
+    }
+}
+
+pub trait BuildCommandExt {
+    fn build(&mut self) -> &mut Self;
+    fn progress(&mut self, progress: Option<Progress>) -> Result<&mut Self>;
+    fn verbose(&mut self, verbosity: Verbosity) -> &mut Self;
+    fn disable_scan_suggest(&mut self) -> &mut Self;
+    fn cross_labels(&mut self, target: &str, platform: &str) -> &mut Self;
+}
+
+impl BuildCommandExt for Command {
+    fn build(&mut self) -> &mut Self {
+        match Engine::has_buildkit() {
+            true => self.args(["buildx", "build"]),
+            false => self.arg("build"),
+        }
+    }
```

Comment:
I think `BuildCommandExt` should be renamed to make clear what it does, e.g invoke the build subcommand

Reply:
Such as `DockerBuildCommentExt`?

Reply:
... I can't think straight :)

the comment was for `BuildCommandExt:build` but I left out `::build` ...

Reply:
maybe

`invoke_build_command`